### PR TITLE
Add job for CI using Prism parser

### PR DESCRIPTION
**Pull request by bquorning**
Inspired by https://github.com/rubocop/rubocop-capybara/pull/112

______________________________________________________________________

Before submitting the PR make sure the following are checked:

- [x] Feature branch is up-to-date with `master` (if not - rebase it).
- [x] Squashed related commits together.
- [ ] Added tests.
- [ ] Updated documentation.
- [ ] Added an entry to the `CHANGELOG.md` if the new code introduces user-observable changes.
- [x] The build (`bundle exec rake`) passes (be sure to run this locally, since it may produce updated documentation that you will need to commit).

### CODE REVIEW

**Comment by pirj** [comment]
But ruby-head is prism already, isn’t this sufficient?

**Comment by bquorning** [comment]
Ah yes, probably so. I'll keep this in draft and look at it again next week.

**Comment by Earlopain** [comment]
My understanding is that it will still use the plain  parser gem, even with ruby-head/ruby-3.4
The prism default is runtime-only. Users need to set it in their config files still, even when targeting ruby 3.3 or higher.

**Comment by pirj** [comment]
Thanks for the hint, @Earlopain !

I’m good with a separate ci job with prism.
Do we need to turn it on in RuboCop somehow? 🤔

**Comment by Earlopain** [comment]
As a user, you would set `AllCops/ParserEngine`. For extensions, this should have you covered: https://github.com/rubocop/rubocop/blob/9f2ffa2270bbc174b2a622350d120b05672a1ce5/lib/rubocop/rspec/cop_helper.rb#L13. I'm not immediatly finding the place where you inherit this, but you probably do somewhere? Since you use the same assertion methods from `RuboCop::RSpec::ExpectOffense`.

For reference, here's the PR for `rubocop-rails` and it basically only sets the env variable. https://github.com/rubocop/rubocop-rails/pull/1245. The other changes are for places that construct `ProcessedSource` themselves but I don't believe that will be the case here.

File: .github/workflows/main.yml [hunk 1]
```diff
@@ -119,3 +119,23 @@ jobs:
...
+          EOF
+      - name: set up Ruby
+        uses: ruby/setup-ruby@v1
+        with:
+          # Specify the minimum Ruby version 2.7 required for Prism to run.
+          ruby-version: "2.7"
+          bundler-cache: true
+      - env:
+          PARSER_ENGINE: parser_prism
+        run: NO_COVERAGE=true bundle exec rake
```
Comment (Earlopain) [inline .github/workflows/main.yml:141]: Docs generator is broken with Ruby 2.7, I openend https://github.com/rubocop/rubocop/pull/13388. Probably, you can just run `bundle exec rake spec` here regardless.

**Comment by bquorning** [comment]
I changed it to run only `rake spec` and not all of `rake`. Does it look ok now?

**Review by ydah — APPROVED** [review]
Looks good! Thank you♥

**Pull request merged**